src/background/messaging/rpc-message-handler.ts

```diff
@@ -1,142 +1,52 @@
 import {
+  type LeatherRpcMethodMap,
   RpcErrorCode,
   type RpcRequests,
   createRpcErrorResponse,
-  getAddresses,
-  open,
-  openSwap,
-  sendTransfer,
-  signMessage,
-  signPsbt,
-  stxCallContract,
-  stxDeployContract,
-  stxGetAddresses,
-  stxSignMessage,
-  stxSignStructuredMessage,
-  stxSignTransaction,
-  stxTransferSip9Nft,
-  stxTransferSip10Ft,
-  stxTransferStx,
-  supportedMethods,
 } from '@leather.io/rpc';
 
 import { queueAnalyticsRequest } from '@background/background-analytics';
-import { rpcSwap } from '@background/messaging/rpc-methods/open-swap';
-import { rpcStxSignTransaction } from '@background/messaging/rpc-methods/stx-sign-transaction';
 
 import { getTabIdFromPort, listenForOriginTabClose } from './messaging-utils';
-import { rpcGetAddresses } from './rpc-methods/get-addresses';
-import { rpcOpen } from './rpc-methods/open';
-import { rpcSendTransfer } from './rpc-methods/send-transfer';
-import { rpcSignMessage } from './rpc-methods/sign-message';
-import { rpcSignPsbt } from './rpc-methods/sign-psbt';
-import {
-  rpcSignStacksMessage,
-  rpcSignStacksStructuredMessage,
-} from './rpc-methods/sign-stacks-message';
-import { rpcStxCallContract } from './rpc-methods/stx-call-contract';
-import { rpcStxDeployContract } from './rpc-methods/stx-deploy-contract';
-import { rpcStxGetAddresses } from './rpc-methods/stx-get-addresses';
-import { rpcStxTransferSip9Nft } from './rpc-methods/stx-transfer-sip9-nft';
-import { rpcStxTransferSip10Ft } from './rpc-methods/stx-transfer-sip10-ft';
-import { rpcStxTransferStx } from './rpc-methods/stx-transfer-stx';
-import { rpcSupportedMethods } from './rpc-methods/supported-methods';
-
-export async function rpcMessageHandler(message: RpcRequests, port: chrome.runtime.Port) {
-  listenForOriginTabClose({ tabId: port.sender?.tab?.id });
-
-  switch (message.method) {
-    case open.method: {
-      await rpcOpen(message, port);
-      break;
-    }
-    case openSwap.method: {
-      await rpcSwap(message, port);
-      break;
-    }
-    case getAddresses.method: {
-      await rpcGetAddresses(message, port);
-      break;
-    }
-
-    case signMessage.method: {
-      await rpcSignMessage(message, port);
-      break;
-    }
-
-    case sendTransfer.method: {
-      await rpcSendTransfer(message, port);
-      break;
-    }
-
-    case signPsbt.method: {
-      await rpcSignPsbt(message, port);
-      break;
-    }
 
-    case stxCallContract.method: {
-      await rpcStxCallContract(message, port);
-      break;
-    }
+type RpcHandler<T> = (request: T, port: chrome.runtime.Port) => Promise<void> | void;
 
-    case stxDeployContract.method: {
-      await rpcStxDeployContract(message, port);
-      break;
-    }
+type RpcHandlers = {
+  [M in keyof LeatherRpcMethodMap]: RpcHandler<LeatherRpcMethodMap[M]['request']>;
```

Comment:
```suggestion
  [Method in keyof LeatherRpcMethodMap]: RpcHandler<LeatherRpcMethodMap[Method]['request']>;
```

Since `M` isn't super common maybe you could just call it `Method` to be explicit?